### PR TITLE
Asynchronous verification of the user.

### LINKED ISSUE #37: Async the hub call for user verification
Currently, the hub is inquired synchronously for the user, meaning that we might get lag if the hub is unreachable. It's unlikely as a scenario, but we prefer to have everything asynchronous.

### PR DESCRIPTION
**Pull request by stefanoborini**
Fixes #37

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -275,6 +279,35 @@ def invocation_argv():
...
+                                        '127.0.0.1',
+                                        family=socket.AF_INET,
+                                        reuse_port=reuse_port)[0]
+    port = sock.getsockname()[1]
+    return sock, port
+
+
+class AsyncHTTPTestCase(tornado.testing.AsyncHTTPTestCase):
+    """Base class workaround for the above condition."""
+    def setUp(self):
```
Comment (kitchoi) [inline tests/utils.py]: Maybe we don't want to overload `setUp` here given we frequently subclass `TempMixin` and use the `TempMixin.setUp` instead.  How about making this a module-level function that takes a test case as an argument?

**Comment by kitchoi** [comment]
Except for one comment above, the rest looks good to me.

File: tests/utils.py (hunk 1 above)
Comment (stefanoborini) [inline tests/utils.py]: I don't understand. the setUp will be called according to the mro...

Comment (kitchoi) [inline tests/utils.py]: Ah right, `TempMixin` calls `super().setUp()` too.  But I saw you did the same `bind_unused_port` handling in `TestHomeHandler.setUp`, so you have set it twice?

File: tests/handlers/test_home_handler.py [hunk 2]
```diff
@@ -46,13 +29,13 @@ def coro(*args, **kwargs):
...
 
-class TestHomeHandler(TempMixin, AsyncHTTPTestCase):
+class TestHomeHandler(TempMixin, utils.AsyncHTTPTestCase):
     def setUp(self):
         self._old_proxy_api_token = os.environ.get("PROXY_API_TOKEN", None)
         os.environ["PROXY_API_TOKEN"] = "dummy_token"
 
         self._bind_unused_port_orig = tornado.testing.bind_unused_port
-        tornado.testing.bind_unused_port = _bind_unused_port
+        tornado.testing.bind_unused_port = utils.bind_unused_port
```
Comment (kitchoi) [inline tests/handlers/test_home_handler.py]: utils.AsyncHTTPTestCase.setUp has this setup already so this is not needed (and line 46 as well)

**Pull request merged**